test/e2e/searchindex_test.go

```diff
@@ -0,0 +1,216 @@
+					Kind:       "AtlasSearchIndexConfig",
+					APIVersion: "atlas.mongodb.com/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-search-index-config",
+					Namespace: testData.InitialDeployments[0].Namespace,
+				},
+				Spec: akov2.AtlasSearchIndexConfigSpec{
+					Analyzer:       pointer.MakePtr("lucene.standard"),
+					SearchAnalyzer: pointer.MakePtr("lucene.standard"),
+				},
+			}
+			Expect(testData.K8SClient.Create(testData.Context, searchIndexConfig)).To(Succeed())
+		})
+
+		By("Creating one search index, type: SEARCH", func() {
+			Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{
+				Name:      testData.InitialDeployments[0].Name,
+				Namespace: testData.InitialDeployments[0].Namespace,
+			}, testData.InitialDeployments[0])).To(Succeed())
+
+			searchIndexesToCreate := []akov2.SearchIndex{
+				{
+					Name:           "test-search-index",
+					Type:           atlasdeployment.IndexTypeSearch,
+					DBName:         DBTraining,
+					CollectionName: DBTrainingCollectionRoutes,
+					Search: &akov2.Search{
+						Mappings: &akov2.Mappings{Dynamic: pointer.MakePtr(true)},
+						SearchConfigurationRef: common.ResourceRefNamespaced{
+							Name:      searchIndexConfig.GetName(),
+							Namespace: searchIndexConfig.GetNamespace(),
+						},
+					},
+				},
+			}
+			testData.InitialDeployments[0].Spec.DeploymentSpec.SearchIndexes = searchIndexesToCreate
+			Expect(testData.K8SClient.Update(testData.Context, testData.InitialDeployments[0])).To(Succeed())
+
+			Eventually(func(g Gomega) {
+				g.Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{
+					Name:      testData.InitialDeployments[0].Name,
+					Namespace: testData.InitialDeployments[0].Namespace,
+				}, testData.InitialDeployments[0])).To(Succeed())
+				g.Expect(len(testData.InitialDeployments[0].Status.SearchIndexes)).To(BeEquivalentTo(1))
+				g.Expect(testData.InitialDeployments[0].Status.SearchIndexes[0].Status).To(BeEquivalentTo(status.SearchIndexStatusReady))
+			}).WithPolling(10 * time.Second).WithTimeout(10 * time.Minute).Should(Succeed())
+		})
+
+		By("Creating one search index, type: VECTOR SEARCH", func() {
+			Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{
+				Name:      testData.InitialDeployments[0].Name,
+				Namespace: testData.InitialDeployments[0].Namespace,
+			}, testData.InitialDeployments[0])).To(Succeed())
+
+			vectorSearchIndexToCreate := akov2.SearchIndex{
+				Name:           "test-search-index",
+				Type:           atlasdeployment.IndexTypeVector,
+				DBName:         DBTraining,
+				CollectionName: DBTrainingCollectionGrades,
+				VectorSearch:   &akov2.VectorSearch{},
+			}
+			testData.InitialDeployments[0].Spec.DeploymentSpec.SearchIndexes = append(
+				testData.InitialDeployments[0].Spec.DeploymentSpec.SearchIndexes, vectorSearchIndexToCreate)
+			Expect(testData.K8SClient.Update(testData.Context, testData.InitialDeployments[0])).To(Succeed())
```

Comment:
we can't do this like this. You have to use https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict using the following idiom:
```
err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
    resource, err := client.Get...
    if err != nil {
        return err
    }

    // Make whatever updates to the resource are needed
    resource.Spec = "foobar"

    // Try to update
    _, err = client.Update...
    return err
})
```